Compilation

matlacki opened the pull request:
Current master branch does not compile. I did numerous fixes to make it compile again. I had to rename DEBUGCODE macro to DEBUGCODE_TT as DEBUGCODE was already defined in t4k_common.h (in latest version pulled from GIT) and there was a conflict. in   src/titlescreen.c     the    DEBUGCODE(debug_all) refers to the macro from t4k_common.h

colors definition was removed from src/titlescreen.c as it is already in t4k_common.h

symbols:

SDL_Thread *tts_thread;
SDL_Thread *tts_announcer_thread;
int  text_to_speech_status=0;

have been defiend in the header files, which led to "multiple defined symbol" during linkin. Now header files have "extern" version of those symbols and new file "globals.c" has been added that hosts the definitions.

With that changes the game compile and runs.

h01ger commented:
could you please explain on which system this is needed?

matlacki commented:
Hi,
It seems my pull requests fixes too many problems. Some of the problems were due to using ubuntu-provided t4k_common.  With t4k-common from git the code actually compiles, but some of the linker problems remain.  I will upload a new pull request that works on "clean" git-based t4k_common

so the answer is this is a good fix if one uses ubuntu version of t4k_common.

h01ger commented:
On Tue, Nov 14, 2023 at 02:07:41AM -0800, Mateusz Łącki wrote:
> It seems my pull requests fixes too many problems. Some of the problems were due to using ubuntu-provided t4k_common.  With t4k-common from git the code actually compiles, but some of the linker problems remain.  I will upload a new pull request that works on "clean" git-based t4k_common

great.
 
> so the answer is this is a good fix if one uses ubuntu version of t4k_common.

I think ubuntu should also use the updated t4k-common :)


-- 
cheers,
	Holger

 ⢀⣴⠾⠻⢶⣦⠀
 ⣾⠁⢠⠒⠀⣿⡁  holger@(debian|reproducible-builds|layer-acht).org
 ⢿⡄⠘⠷⠚⠋⠀  OpenPGP: B8BF54137B09D35CF026FE9D 091AB856069AAA1C
 ⠈⠳⣄

If you’re going through hell, keep going!

h01ger commented:
On Tue, Nov 14, 2023 at 02:57:28AM -0800, Holger Levsen wrote:
> > so the answer is this is a good fix if one uses ubuntu version of t4k_common.
> I think ubuntu should also use the updated t4k-common :)

so probably the way to go is to first do a new t4k-common release,
and then work on fixing tuxtype...


-- 
cheers,
	Holger

 ⢀⣴⠾⠻⢶⣦⠀
 ⣾⠁⢠⠒⠀⣿⡁  holger@(debian|reproducible-builds|layer-acht).org
 ⢿⡄⠘⠷⠚⠋⠀  OpenPGP: B8BF54137B09D35CF026FE9D 091AB856069AAA1C
 ⠈⠳⣄

Life is short but a sea of morons is forever.

matlacki commented:
Hi,
I guess so. In any case I made a new pull request (number 36). Some of the problems I fixed (all compilation problems) in this pull request were related to the old t4k_common in ubuntu.  I guess the value of this pull request #32 is approaching zero...

Now the new pull request is based on t4k_common from git. 
So I have removed ubuntu version completely to be 100% sure I do not use both libraries at some point, and I just use the self-compiled t4k from git  (to build it the https://github.com/tux4kids/t4kcommon/pull/10 was actually helpful as there were some bugs as well)

With that version of t4k_common, the the tuxtype compiles well, but linker complains about "multiply defined symbols".

In tuxtype source tree there is essentially one simple bug but repeated few times with respect to 3-4 symbols: a symbol gets defined and declared in .h file, which is included in multiple *.c files, this leads to "multiply defined symbol".

Best,
Mateusz

h01ger commented:
On Tue, Nov 14, 2023 at 03:19:50AM -0800, Mateusz Łącki wrote:
> I guess so. In any case I made a new pull request (number 36). Some of the problems I fixed (all compilation problems) in this pull request were related to the old t4k_common in ubuntu.  I guess the value of this pull request #32 is approaching zero...

ok, so let's close #32 then?
 
> Now the new pull request is based on t4k_common from git. 
> So I have removed ubuntu version completely to be 100% sure I do not use both libraries at some point, and I just use the self-compiled t4k from git  (to build it the https://github.com/tux4kids/t4kcommon/pull/10 was actually helpful as there were some bugs as well)

cool!

> With that version of t4k_common, the the tuxtype compiles well, but linker complains about "multiply defined symbols".

one step forward... :)

> In tuxtype source tree there is essentially one simple bug but repeated few times with respect to 3-4 symbols: a symbol gets defined and declared in .h file, which is included in multiple *.c files, this leads to "multiply defined symbol".

*nods*

-- 
cheers,
	Holger

 ⢀⣴⠾⠻⢶⣦⠀
 ⣾⠁⢠⠒⠀⣿⡁  holger@(debian|reproducible-builds|layer-acht).org
 ⢿⡄⠘⠷⠚⠋⠀  OpenPGP: B8BF54137B09D35CF026FE9D 091AB856069AAA1C
 ⠈⠳⣄

Dat gifft in Plattdüütschen keen Woort för „Flüchtlinge”. Dat sünd allens Lüüd, 
Mischen, Kinners, Olle, Froons, Mannslüüd, so as Du un Ick.

matlacki commented:
Yes, it can be closed.
